hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/ComplexKeyGenerator.java

```diff
@@ -64,6 +63,9 @@ public String getPartitionPath(GenericRecord record) {
 
   @Override
   public String getRecordKey(Row row) {
+    if (recordKeyFields.isEmpty()) {
+      return EMPTY_STRING;
```

Comment:
I'd suggest create a method `emptyKey` and redirect there so that if we decided to change the sentinel value we can switch it easily